### PR TITLE
java: remove code_size argument from execute()

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@atoulme @jrhea this is another potential place where buffer overread could happen, the unchecked copying prior to this PR.

File: bindings/java/c/evmc-vm.c [hunk 1]
```diff
@@ -90,12 +90,12 @@ JNIEXPORT void JNICALL Java_org_ethereum_evmc_EvmcVm_execute(JNIEnv* jenv,
                                                              jint jrev,
                                                              jobject jmsg,
                                                              jobject jcode,
-                                                             jint jcode_size,
                                                              jobject jresult)
 {
     (void)jcls;
     struct evmc_message* msg = (struct evmc_message*)(*jenv)->GetDirectBufferAddress(jenv, jmsg);
     assert(msg != NULL);
+    size_t code_size = (size_t)(*jenv)->GetDirectBufferCapacity(jenv, jcode);
```
Comment (chfast) [inline bindings/java/c/evmc-vm.c:98]: Add assert that `code_size` is not negative. Probably should be done before casting to `size_t`.

**Review by chfast — CHANGES_REQUESTED** [review]

File: bindings/java/c/evmc-vm.c (hunk 1 above)
Comment (axic) [reply bindings/java/c/evmc-vm.c]: Done.

**Review by chfast — APPROVED** [review]

**Review by chfast — APPROVED** [review]

File: bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java [hunk 2]
```diff
@@ -114,11 +113,11 @@ native void execute(
    * <p>This allows the context to managed in one method
    */
   public synchronized ByteBuffer execute(
-      HostContext context, int rev, ByteBuffer msg, ByteBuffer code, int size) {
```
Comment (axic) [inline bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java:117]: Note: this is a breaking change, but the Java binding we have is not stable enough and more changes are coming.

Everything merged so far were non-breaking changes.

**Pull request merged**